src/Models/Dto/ActionPlanDto.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Middleware.Models.Domain;
+using Redis.OM.Modeling;
+
+namespace Middleware.Models.Dto;
+
+[Document(IndexName = "actionPlan-idx", StorageType = StorageType.Json, Prefixes = new[] { "ActionPlan" })]
+public class ActionPlanDto : Dto
+{
+    [Indexed]
+    [RedisIdField]
+    public override string Id { get; set; }
+
+    [Indexed]
+    public string? TaskId { get; set; }
+    [Indexed]
+    public string Name { get; set; }
+    [Indexed]
+    public string? Status { get; set; }
+    [Indexed]
+    public bool IsReplan { get; set; }
+    [Indexed(Sortable = true)]
+    public DateTime LastStatusChange { get; set; }
+    [Indexed]
+    public List<ActionDto> ActionSequence { get; set; }
```

Comment:
For the purpose of storing the actual status of the deployed ActionPlan we will have to store ActionModels instead of ActionDto

Reply:
Conflict with other parts of the code, needs review